Kernel/Arch/aarch64/PowerState.cpp

```diff
@@ -5,19 +5,25 @@
  */
 
 #include <Kernel/Arch/PowerState.h>
+#include <Kernel/Arch/aarch64/PSCI.h>
 #include <Kernel/Arch/aarch64/RPi/Watchdog.h>
 #include <Kernel/Firmware/DeviceTree/DeviceTree.h>
 
 namespace Kernel {
 
 void arch_specific_reboot()
 {
+    if (PSCI::is_supported())
+        PSCI::reset();
 }
 
 void arch_specific_poweroff()
 {
     if (DeviceTree::get().is_compatible_with("raspberrypi,3-model-b"sv) || DeviceTree::get().is_compatible_with("raspberrypi,4-model-b"sv))
         RPi::Watchdog::the().system_shutdown();
+
+    if (PSCI::is_supported())
```

Comment:
I would try `PSCI::poweroff` before platform-specific mechanisms, as the firmware might do some extra stuff before shutdown.